regression/goto-harness-gb-input/Makefile

```diff
@@ -0,0 +1,18 @@
+default: tests.log
```

Comment:
:question: How come we need to introduce a whole new test directory?

Reply:
The reason for this is that the other ones do not have the same compile->test pipeline.

The `goto-harness` directory takes as input `.c` files, passes them to `goto-cc`, gets a binary out, and then feeds that to `goto-harness`.

In this directory I just pass a goto binary directly to `goto-harness`, and the result of that straight into cbmc to make sure there's no conversion error (i.e. erroneous output in the c file produced by the harness).

Reply:
Oh I see, meaning you're checking in goto binary files. Do you have to? Can you not run the C file through goto-cc as part of the test - checking in binary files is a recipe for them going out of date when someone changes the C file but doesn't realize they need to run goto-cc manually.

Reply:
This is a test on `goto-harness` only: We don't need a `.c` file because the input to `goto-harness` is a goto-binary (hence why it's unneeded to run `goto-cc`).

Also, the reason this is based on a binary test is that because this was very hard to reproduce based on a c file (as this was based on the compilation pipeline of the aws-c-common repository), I figured I could only use the binary (which is the output of the `make goto` step), which is what I need to feed to `goto-harness` for it to manifest the problem.

So in a sense, we don't run the risk of it getting stale (unless we change the internal representation of goto-binaries) and the binary attached becomes stale - in which case it's easy enough to reproduce.

Reply:
Hmm.. I didn't initially spot the source C file is missing. :no_entry_sign: At a min, there needs to be a readme of how someone would go about recreating this. I'd rather a little more work on coming up with a simpler reproducing example (didn't we have one that admittedly relied on using goto-cc, but at least was simpler). This will also I guess fix the windows CI problem

Reply:
Yes; binary without source and build instructions is a hard block.  Even if it is not what the test is for, I would *really* prefer it if the test compiled up the goto-binaries it needs.